guestlink.php

```diff
@@ -0,0 +1,90 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Page to grant external users access to a BBB session
+ *
+ * @package    mod_bigbluebuttonbn
+ * @author     Angela Baier
+ * @copyright  2020 University of Vienna
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require('../../config.php');
+require_once(__DIR__ . '/lib.php');
+require_once(__DIR__ . '/locallib.php');
+
+global $PAGE, $OUTPUT;
+
+$gid = required_param('gid', PARAM_ALPHANUM); // This is required.
+$guestname = optional_param('guestname', '', PARAM_TEXT);
+$guestpass = optional_param('guestpass', '', PARAM_TEXT);
+$PAGE->set_url(new moodle_url('/mod/bigbluebuttonbn/guestlink.php',
+        ['gid' => $gid, 'guestname' => $guestname, 'guestpass' => $guestpass]));
+$PAGE->set_context(context_system::instance());
+$PAGE->set_pagelayout('standard');
+
+if (!\mod_bigbluebuttonbn\locallib\config::get('participant_guestlink')) {
+    echo get_string('guestlink_form_guestlink_disabled', 'bigbluebuttonbn');
+    die();
+}
+
+$bigbluebuttonbn = bigbluebuttonbn_get_bigbluebuttonbn_by_guestlinkid($gid);
+if (!$bigbluebuttonbn->guestlinkenabled) {
+    echo get_string('guestlink_form_guestlink_disabled_instance', 'bigbluebuttonbn');
+    die();
+}
+$valid = ($guestname && ($bigbluebuttonbn->guestpass == $guestpass || !$bigbluebuttonbn->guestpass));
+
+if (!$valid) {
+    $guestpasserrormessage = false;
+    $guestnameerrormessage = false;
+    if ($guestpass && $bigbluebuttonbn->guestpass != $guestpass) {
+        $guestpasserrormessage = true;
+    }
+    $context = ['name' => $bigbluebuttonbn->name, 'gid' => $gid,
+        'guestpassenabled' => $bigbluebuttonbn->guestpass,
+        'guestpasserrormessage' => $guestpasserrormessage,
+        'guestnameerrormessage' => $guestnameerrormessage,
+        'guestname' => $guestname,
+    ];
+
+    echo $OUTPUT->header();
+    echo $OUTPUT->render_from_template('mod_bigbluebuttonbn/guestaccess_view', $context);
```

Comment:
We've done some internal testing with a real client and functionally its pretty good but most of the feedback is minor styling and ux.

For instance was there a good reason to do this via a custom template and not using formslib?

I appreciate that this makes it easier to override, but I'm not sure that many people will, and it looks different enough to normally styled forms that it looks out of place.

In a similar way, why is there ```<div class="warning">``` inside the mustache template with custom css, this really should just leverage the existing renderers for this, specifically like:

```
$OUTPUT->notification(get_string('plugin', 'error'), \core\output\notification::NOTIFY_ERROR);
```

https://github.com/catalyst/moodle-mod_bigbluebuttonbn/pull/4/files#diff-380b7b38760dd442e897eb0164c58f6a17da966ccaca6318017a468c163979b1R69